scripts/release.js

```diff
@@ -55,13 +56,28 @@ const archive = async() => {
   );
 };
 
+const writeSegmentFile = () => {
+  console.log(process.env.SEGMENT_API_KEY)
+  const key = `module.exports = SEGMENT_API_KEY = '${process.env.SEGMENT_API_KEY}';`;
+  // create directly in cli-repl/lib so it can be part of artifacts in dist
+  const configPath = path.join(__dirname, '..', 'packages', 'cli-repl', 'lib', 'config.js');
+  try {
+    // can just write this sync, since it's part of the release script
+    fs.writeFileSync(configPath, key)
+  } catch (e) {
+    console.log('mognosh: unable to write segment api key')
+    return;
+  }
+}
+
 /**
  * Creates dist/mongosh or dist/mongosh.exe and then creates a
  * tarball or zip of the executable as dist/mongosh-${version}-${os}.tgz or
  * zip.
  */
 const release = async() => {
   const artifact = getArtifact();
+  const segmentFile = await writeSegmentFile();
```

Comment:
I think this `segmentFile` variable can be removed here, seems unused. Also the `await` is  probably unnecessary at this point.